jwst/regtest/regtestdata.py

```diff
@@ -52,18 +53,30 @@ def __init__(self, env="dev", inputs_root="jwst-pipeline",
         self.remote_results_path = remote_results_path
         self.test_name = test_name
         self.traceback = traceback
+        self.okify_op = okify_op
 
         # Initialize non-initialized attributes
         self.asn = None
 
-    def __repr__(self):
-        return pprint.pformat(
-            dict(input=self.input, output=self.output, truth=self.truth,
-                 input_remote=self.input_remote, truth_remote=self.truth_remote,
-                 remote_results_path=self.remote_results_path, test_name=self.test_name,
-                 traceback=self.traceback),
-            indent=1
-        )
+    @classmethod
+    def open(cls, filename):
+        with asdf.open(filename) as af:
+            return cls(**af.tree)
+
+    @property
+    def bigdata_root(self):
+        return self._bigdata_root
+
+    @property
+    def input(self):
+        return self._input
+
+    @input.setter
+    def input(self, value):
+        if value:
+            self._input = os.path.abspath(value)
+        else:
+            self._input = value
```

Comment:
Ok, this diff is impossible to follow, as I assume you've changed things, but since you've also re-ordered the methods, it's impossible to follow.  Could you keep the order the same, so it is clear to see what has changed?

Feel free to reorder them in a follow-up PR.

Reply:
✅